lib/puppet/type/user.rb

```diff
@@ -171,10 +171,15 @@ def sync
 
     newproperty(:comment) do
       desc "A description of the user.  Generally the user's full name."
-      if RUBY_VERSION < "2.1.0"
-        munge do |v|
-          v.respond_to?(:force_encoding) ? v.force_encoding(Encoding::ASCII_8BIT) : v
+      def insync?(current)
+        # nameservice provider requires special attention to encoding
+        # Use Puppet::Property#should instead of @should to avoid extra
+        # array handling logic
+        if !@should.empty? and provider.respond_to?(:comments_insync?)
```

Comment:
I agree most implementations just pass `@should` value. I was certainly on the fence about that - not sure which way was better. Happy to update to handle accepting `@should`. In that case, I believe the value we're going to get passed in will almost always be an array containing a single string representing the comment value.